Sources/StreamChat/Database/DTOs/MemberModelDTO.swift

```diff
@@ -130,11 +130,20 @@ extension NSManagedObjectContext {
 }
 
 extension MemberDTO {
-    func asModel() -> ChatChannelMember { .create(fromDTO: self) }
+    func asModel() -> ChatChannelMember? { .create(fromDTO: self) }
 }
 
 extension ChatChannelMember {
-    fileprivate static func create(fromDTO dto: MemberDTO) -> ChatChannelMember {
+    fileprivate static func create(fromDTO dto: MemberDTO) -> ChatChannelMember? {
+        if dto.isDeleted {
+            log.warning("Cannot create an instance of MemberDTO id: \(dto.id), the DTO was marked as deleted")
+            return nil
+        }
+
+        guard let user = dto.user.asModel() else {
+            return nil
+        }
```

Comment:
Some here, we can have a `ChatChannelMember` without a `user`. Ideally `dto.user` should be optional